### PR TITLE
Fix/disable new room non workspace members

### PR DESCRIPTION
**Pull request by mananjadhav**
### Details

- New Room by default requires Workspace Selection in the dropdown. Hence, if the user doesn't have any workspace created then we remove the "New Room" menu from the Action Menu.

### Fixed Issues

$ https://github.com/Expensify/App/issues/7668

### Tests
1. Tested with workspace to see the new room
2. Tested without workspace to check "New Room" doesn't show up.

- [x] Verify that no errors appear in the JS console

### QA Steps

As an account that is not a member of any workspace:

1. Click the "+" global create icon
2. Check that "New Room" doesn't show up

- [x] Verify that no errors appear in the JS console

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web

web-new-room-enabled
web-new-room-disabled

#### Mobile Web

mweb-new-room-disabled

#### Desktop

desktop-new-room-disabled

#### iOS

ios-new-room-disabled

#### Android

android-new-room-disabled

### CODE REVIEW

File: src/pages/home/sidebar/SidebarScreen.js [hunk 1]
```diff
@@ -122,6 +122,7 @@ class SidebarScreen extends Component {
     }
 
     render() {
+        const workspaces = _.filter(this.props.allPolicies, policy => policy && policy.type === CONST.POLICY.TYPE.FREE);
```
Comment (parasharrajat) [inline src/pages/home/sidebar/SidebarScreen.js:125]: I see this logic is used in multiple places. it is not clear why we are filtering the `CONST.POLICY.TYPE.FREE` policies unless we have a comment. 

So  better would be create a action method in action/Policy.js `getWorkspacePolicies(policies: Array<Policy>): Array<WorkspacePolicy>`. Which does the same thing but you leave a comment the same as on the New Room Page and then replace all the places where we are getting workspaces from policies with this method call.

**Review by parasharrajat — CHANGES_REQUESTED** [review]

File: src/pages/home/sidebar/SidebarScreen.js (hunk 1 above)
Comment (mananjadhav) [reply src/pages/home/sidebar/SidebarScreen.js:125]: I did think about moving it to action but wasn't sure. 

https://github.com/Expensify/App/blob/d9355db0e9685021c3be1debca3c43751ad2d759/src/pages/workspace/WorkspaceNewRoomPage.js#L61-L65

`WorkspaceNewRoomPage` is where we have the same code, and a small comment which says `Workspaces are policies with type === 'free'`.

What happens is, even for a new user a policy exist of `type === personal`, with a default name like "Manan jadhav's expenses"

Comment (parasharrajat) [reply src/pages/home/sidebar/SidebarScreen.js:125]: Maybe just leave it for now and add a small comment. That should be enough. There are other places where we check the free policy such as InitialSettingsPage. But adding another filter call would be extra thing to do as we can filter the list based on multiple things. 

So Just a comment should be enough..

Comment (mananjadhav) [reply src/pages/home/sidebar/SidebarScreen.js:126]: I've added the comment

**Review by parasharrajat — DISMISSED** [review]
LGTM.

cc: @NikkiWines 

:ribbon: :eyes: :ribbon:  C+ reviewed

File: src/pages/home/sidebar/SidebarScreen.js [hunk 2]
```diff
@@ -160,7 +162,7 @@ class SidebarScreen extends Component {
                                     text: this.props.translate('sidebarScreen.newGroup'),
                                     onSelected: () => Navigation.navigate(ROUTES.NEW_GROUP),
                                 },
-                                ...(Permissions.canUsePolicyRooms(this.props.betas) ? [
+                                ...(Permissions.canUsePolicyRooms(this.props.betas) && workspaces.length > 0 ? [
```
Comment (NikkiWines) [inline src/pages/home/sidebar/SidebarScreen.js:165]: You can do just `workspaces.length` here I believe.

**Review by NikkiWines — CHANGES_REQUESTED** [review]
one minor minor coment but otherwise looks great

File: src/pages/home/sidebar/SidebarScreen.js (hunk 2 above)
Comment (mananjadhav) [reply src/pages/home/sidebar/SidebarScreen.js]: Updated

**Review by NikkiWines — DISMISSED** [review]
👍

**Comment by NikkiWines** [comment]
hmm, not sure why the linter is failing. Maybe try merging main into your branch @mananjadhav

**Comment by mananjadhav** [comment]
@NikkiWines Done

**Review by NikkiWines — APPROVED** [review]
👍

**Comment by Stutikuls** [comment]
**Issue 1 -**

**Title**- [Medium] Chrome +Jaws : 'Room Name' & 'Workspace' controls are not marked as required and announced.
**Actual** - 'Room Name' & 'Workspace' fields are required to be filled but not announced as required and no visual "*" sign for required field. 
**Expected** -If the form element is required, it should be announced as such for the screen reader users and visual "*" sign for required field should be there. 
**WCAG failure** - 1.3.1
**Reproducible in staging?**: - Yes
**Version Number**: - v1.1.42-2
**Platforms** - Web (Chrome + Jaws), Desktop (MAC), iOS(Voiceover), Mobile - web(Safari + Voiceover)

**Note** - Due to installation of build not able to check on Android

https://user-images.githubusercontent.com/97600950/157794089-3e50e37a-5f14-4bc2-ac98-8696ae030d84.mp4

https://user-images.githubusercontent.com/97600950/157184775-555e29e3-b3b8-484d-975b-253176be91a4.mp4


**Issue 2 -** 

**Title**-[Medium] Chrome + Jaws: Screen reader : Error message is not announced by the screen reader.
**Actual** - Screen reader does not announce any inline error message appear below to text boxes.
**Expected** - Inline error message should be conveyed to the users as soon as it appear on the fields so that user can resolve the error.
**WCAG failure** - 3.3.1
**Reproducible in staging?**: - Yes
**Version Number**: - v1.1.42-2
**Platforms** - Web(Chrome + Jaws), Desktop(MAC), Mobile-web (Safari + Voiceover), iOS(Voiceover)

**Note** - Due to installation of build not able to check on Android

https://user-images.githubusercontent.com/97600950/157794231-e305ad95-f0e9-41a9-9934-b3c0cf45afae.mp4

**Issue 3 -** 

**Title**-[Medium] Chrome + Jaws : Screen reader : Screen reader is not reading role for 'create room' control.
**Actual** - Screen reader only announce "create room".
              **Note -** On Mac reading "create room Group" when focus lands on the create room control.
**Expected** - Role should define as 'Button' and screen reader should read like "Create room button. 
**WCAG failure** - 4.1.2
**Reproducible in staging?**: - Yes
**Version Number**: - v1.1.42-2
**Platforms** - Web (Chrome + Jaws), Desktop(MAC), Mobile-web (Safari + Voiceover)

**Note** - Due to installation of build not able to check on Android

https://user-images.githubusercontent.com/97600950/157795064-fabf495d-a27f-414f-98b9-831d1b6a77db.mp4

**Issue 4 -** 

**Title**-[Medium] Chrome : 'No workspace' option is showing on the 'New(+)' popup options.
**Actual** - 'No workspace' option is showing on the 'New(+)' popup options.
             **Note-** only showing on MAC.
**Expected** - 'No workspace' option  showed show on the 'New(+)' popup options.
**WCAG failure** - Nil
**Reproducible in staging?**: - Yes
**Version Number**: - v1.1.42-2
**Platforms** - Web (Chrome + Jaws), Mobile-web (Safari + Voiceover)

**Note** - Due to installation of build not able to check on Android
7829_No workspace is showing

**Comment by NikkiWines** [comment]
@Stutikuls for Issue 4 ` 'No workspace' option is showing on the 'New(+)' popup options.`, can you confirm that the account does not have any existing workspaces? Also, there is not a "No workspace" option, but there is a "New Workspace". Is this the option you mean? 

@mananjadhav the other issues that @Stutikuls reported all look good and should be fixed.

**Comment by mananjadhav** [comment]
@NikkiWines Sure I can fix them, are you suggesting it is a part of #7668?

The reason I am asking is that these issues existed before my change. In addition to this, my changes were not related to the form, but the options in the FAB menu. So I am not sure if it would be a part of the current PR.

**Comment by NikkiWines** [comment]
Oh, good point - sorry, @mananjadhav. @Stutikuls can you make one new issue for issues 1 through 3 as they're not related to this specific PR.

**Comment by Stutikuls** [comment]
> @Stutikuls for Issue 4 ` 'No workspace' option is showing on the 'New(+)' popup options.`, can you confirm that the account does not have any existing workspaces? Also, there is not a "No workspace" option, but there is a "New Workspace". Is this the option you mean?
> 
> @mananjadhav the other issues that @Stutikuls reported all look good and should be fixed.

@NikkiWines I mean, No Workspace option is not appearing after expand the 'New(+) button.

**Comment by Stutikuls** [comment]
> Oh, good point - sorry, @mananjadhav. @Stutikuls can you make one new issue for issues 1 through 3 as they're not related to this specific PR.

@NikkiWines Log it in the same PR as a new bug or some other place need to log as a new?

**Comment by NikkiWines** [comment]
>  I mean, No Workspace option is not appearing after expand the 'New(+) button.

Ah, ok. Does the account you're using to test this have any existing workspaces? 

> Log it in the same PR as a new bug or some other place need to log as a new?

For issues 1-3 we'd want to create a new issue (following [this template](https://github.com/Expensify/App/issues/new?assignees=&labels=&template=Accessibility.md)) since they are new bugs that aren't related to this PR or the issue it resolves.

**Comment by Christinadobrzyn** [comment]
Just checking are we waiting on a fix for Issue 4 in this PR? Not sure if I should if https://github.com/Expensify/App/issues/7668 can be paid?

**Comment by mananjadhav** [comment]
@Christinadobrzyn I don’t think anything is pending for this PR

**Comment by NikkiWines** [comment]
Yes, @Christinadobrzyn - there are some new accessibility issues but we're waiting on some more information since 3 of them are unrelated to this PR and we're waiting on more information for Issue 4: `'No workspace' option is showing on the 'New(+)' popup options.` though I've just realized that I can test this locally myself.

**Comment by NikkiWines** [comment]
1. **Account With Existing Workspace** - `New Workspace` option does **not** show in the FAB menu, as expected ✅ 

| Account Details | FAB Menu|
| ------------- | ------------- |
| image | image |

2. **Account Without Workspaces** - `New Workspace` option is shown in the FAB menu, as expected ✅ 

| Account Details | FAB Menu|
| ------------- | ------------- |
| Screen Shot 2022-03-18 at 09 59 09 | image |

So yes, I think this PR is fine and we should issue payment. The accessibility issues 1-3 should be put in a new issue because they're unrelated to this PR or the issue this PR resolves.

**Pull request merged**